content/automate/ManageIQ/Transformation/Common.class/__methods__/assesstransformation.rb

```diff
@@ -13,6 +13,35 @@ def initialize(handle = $evm)
             @handle = handle
           end
 
+          def network_mappings(task, vm)
+            mappings = []
+            vm.hardware.nics.select { |n| n.device_type == 'ethernet' }.each do |nic|
+              source_network = nic.lan
+              destination_network = task.transformation_destination(source_network)
+              raise "[#{vm.name}] NIC #{nic.device_name} [#{source_network.name}] has no mapping. Aborting." if destination_network.nil?
+              mappings << {
+                :source      => source_network.name,
+                :destination => destination_network.name,
+                :mac_address => nic.address
+              }
+            end
+            mappings
+          end
+
+          def storage_mappings(task, vm)
+            vm.hardware.disks.select { |d| d.device_type == 'disk' }.each do |disk|
```

Comment:
@fdupont-redhat 
Either you can replace the each with collect and remove the appending into virtv2v_disks or initialize virtv2v_disks array like you do for mapping

```
def storage_mappings(task, vm)
            vm.hardware.disks.select { |d| d.device_type == 'disk' }.collect do |disk|
source_storage = disk.storage
              destination_storage = task.transformation_destination(disk.storage)
              raise "[#{vm.name}] Disk #{disk.device_name} [#{source_storage.name}] has no mapping. Aborting." if destination_storage.nil?
             {
                :path    => disk.filename,
                :size    => disk.size,
                :percent => 0,
                :weight  => disk.size.to_f / vm.allocated_disk_storage.to_f * 100
              }
            end
          end
```